superbench/benchmarks/micro_benchmarks/ib_loopback_performance.py

```diff
@@ -146,33 +150,39 @@ def _preprocess(self):
                     )
                 )
                 return False
-            else:
-                try:
-                    command = os.path.join(self._args.bin_dir, self._bin_name)
-                    numa_cores = get_numa_cores(self._args.numa)
-                    if len(numa_cores) < 2:
-                        self._result.set_return_code(ReturnCode.MICROBENCHMARK_DEVICE_GETTING_FAILURE)
-                        logger.error('Getting numa core devices failure - benchmark: {}.'.format(self._name))
-                        return False
-                    if len(numa_cores) >= 4:
-                        server_core = int(numa_cores[-1])
-                        client_core = int(numa_cores[-3])
-                    else:
-                        server_core = int(numa_cores[-1])
-                        client_core = int(numa_cores[-2])
-                    command += ' ' + str(server_core) + ' ' + str(client_core)
-                    command += ' ' + os.path.join(self._args.bin_dir, self.__support_ib_commands[ib_command])
-                    command += command_mode + ' -F'
-                    command += ' --iters=' + str(self._args.iters)
-                    command += ' -d ' + network.get_ib_devices()[self._args.ib_index].split(':')[0]
-                    command += ' -p ' + str(network.get_free_port())
-                    command += ' -x ' + str(self._args.gid_index)
-                    command += ' --report_gbits'
-                    self._commands.append(command)
-                except BaseException as e:
-                    self._result.set_return_code(ReturnCode.MICROBENCHMARK_DEVICE_GETTING_FAILURE)
-                    logger.error('Getting ib devices failure - benchmark: {}, message: {}.'.format(self._name, str(e)))
-                    return False
+
+            try:
+                self.__sock_fds.append(socket.socket(socket.AF_INET, socket.SOCK_STREAM))
```

Comment:
Do we need to keep all __scok_fds after it has been reused?

Reply:
this fix is to bind these sockets before running perftest process and keep them until perftest finishes

Reply:
Yes, my question is the purpose to keep them because it's like just to occupy the port before the test. For example, if the perftest will take 2 hours. We need to keep double the connect socket for a long time. Each socket in some scenario also be set to a larger buffer due to Internet/DC requirement (like BRB, DCTCP), then it will be very large buffer cost (1MB-4MB for one socket). Is there any way not to keep it?

Reply:
If it's too complicated, we can keep this.

Reply:
Thanks for explanation